client/my-sites/marketplace/pages/marketplace-product-install/index.tsx

```diff
@@ -272,9 +275,33 @@ const MarketplaceProductInstall = ( {
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [ pluginUploadComplete, installedPlugin, setCurrentStep ] );
 
-	const pluginsUrl = useSelector( ( state ) =>
+	// Fetch fresh site data (including admin_url) post-transfer via TanStack Query
+	const { data: freshSite } = useQuery( {
+		...siteByIdQuery( siteId ?? 0 ),
+		enabled: !! siteId && ( ! atomicFlow || automatedTransferStatus === transferStates.COMPLETE ),
+		refetchInterval: ( query ) => {
+			if ( ! query.state.data ) {
+				return 0;
+			}
```

Comment:
Won't this prevent further polling? What I mean is that if for some reason the data returned is missing, it won't fetch anymore.

But maybe I missed something.

I think we could also return `false` instead of `0` to stop fetching.

Reply:
I agree that we might need to poll here at all. AFAIK after the `automatedTransferStatus === transferStates.COMPLETE` condition in the `enabled` property is true, this request _should_ return the updated site URL and the `is_wpcom_atomic` flag set to true.

If this is isn't always the case, then we still need to poll here, but in that case I think we should return the same interval (2000) if there's no data yet.